### PR TITLE
Add unit tests for clustering

### PR DESCRIPTION
**Pull request by Smrtnyk**
I will mark this as draft until I push some unit tests (already have some created locally
Currently I added the jest setup and types for clusters (took me a while to figure them out)

### CODE REVIEW

File: src/plugins/utils/tree-clusters.ts [hunk 1]
```diff
@@ -88,30 +100,28 @@ export const metaClusterizeFlatTree = (flatTree: FlatTree, condition = defaultCl
         .filter((nodes) => nodes.length)
         .map((nodes) => ({
             nodes,
-            parents: [...new Set(nodes.map(({ parent }) => parent))],
```
Comment (Smrtnyk) [inline src/plugins/utils/tree-clusters.ts:91]: this thing seems like it was never used?

**Comment by Smrtnyk** [comment]
Not the most ground breaking tests but it is a start
also adds proper types for clusters (which is more progress towards strict types)
I will be on vacation from tomorrow, but somewhere next week I might continue work on stricter types

File: src/plugins/utils/tree-clusters.spec.ts [hunk 2]
```diff
@@ -0,0 +1,797 @@
...
+                type: 'wnTOxfON',
+                children: [],
+                level: 1,
+                index: 4,
+            },
+        ],
+    },
+];
+
+describe('tree clusters', () => {
```
Comment (pyatyispyatil) [inline src/plugins/utils/tree-clusters.spec.ts:781]: Might be preferable to describe smaller data for tests? I think it's too hard to maintain. Also, it would be nice to be able to add tests for edge cases, and without a simple structure, this is very difficult to do. But for first time it is ok.)

File: src/plugins/utils/tree-clusters.ts (hunk 1 above)
Comment (pyatyispyatil) [reply src/plugins/utils/tree-clusters.ts:91]: Well, it looks like that. I searched to see if there is a use for this variable somewhere and did not find it.)

**Review by pyatyispyatil — APPROVED** [review]

**Comment by pyatyispyatil** [comment]
Sorry for delay. I got stuck in work.)
About PR - good start for tests. This should be a nice stability improvement for our development.)
And I have good news - soon I may be spending some of my working time on this project, and it should be on a permanent basis.)

File: src/plugins/utils/tree-clusters.spec.ts (hunk 2 above)
Comment (Smrtnyk) [reply src/plugins/utils/tree-clusters.spec.ts:781]: yeah, tests need to be more refined here
for now, they serve as some kind of protection against regressions, but not fully
I am back from vacation, so might work on that next these days to come up with something more maintainable with more coverage

Comment (pyatyispyatil) [reply src/plugins/utils/tree-clusters.spec.ts:781]: By the way, have you begun to clearly understand the whole project and clustering in particular? I think it will help you in future work with flame-chart.)

Comment (pyatyispyatil) [reply src/plugins/utils/tree-clusters.spec.ts:781]: Yes, you're right.) Good to hear. 

Cool, I think you will soon understand what you want. You can also write something about the plugins you want to implement in the roadmap.

Comment (Smrtnyk) [reply src/plugins/utils/tree-clusters.spec.ts:781]: Currently at my work we have a sidebar with all the data as a list, and when we click on one of the entry it shows in the list only those with that name and highlights all of the nodes in the flamechart also
we don't have the outline, what we do we whiten out all of the nodes and preserve the color only on nodes that are selected
then when we select a specific one from that list, flamechart zooms in and whitens out everything except that one in the list that is clicked. Same is the other way around, if you click on the node in flamechart, it shows you in the list.
tracer-all-selected

Not sure if that is feasible currently as a  custom made plugin passed in from outside to a flamechart
